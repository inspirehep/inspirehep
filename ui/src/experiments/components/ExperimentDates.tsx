import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import { InlineUL } from '../../common/components/InlineList';
import { hasDayMonthAndYear, hasMonthAndYear } from '../../common/utils';

function getDisplayFormatForDateString(date) {
  if (hasDayMonthAndYear(date)) {
    return 'MMM D, YYYY';
  }

  if (hasMonthAndYear(date)) {
    return 'MMM, YYYY';
  }

  return 'YYYY';
}

function getFormattedDate(date) {
  return moment(date).format(getDisplayFormatForDateString(date));
}

function ExperimentDates({
  dateStarted,
  dateProposed,
  dateApproved,
  dateCompleted,
  dateCancelled,
  wrapperClassName,
}) {
  return (
    <InlineUL wrapperClassName={wrapperClassName}>
      {dateProposed && <span>Proposed: {getFormattedDate(dateProposed)}</span>}
      {dateApproved && <span>Approved: {getFormattedDate(dateApproved)}</span>}
      {dateStarted && <span>Started: {getFormattedDate(dateStarted)}</span>}
      {dateCancelled && (
        <span>Cancelled: {getFormattedDate(dateCancelled)}</span>
      )}
      {dateCompleted && (
        <span>Completed: {getFormattedDate(dateCompleted)}</span>
      )}
      {!dateCancelled && !dateCompleted && <span>Still Running</span>}
    </InlineUL>
  );
}

ExperimentDates.propTypes = {
  dateStarted: PropTypes.string,
  dateProposed: PropTypes.string,
  dateApproved: PropTypes.string,
  dateCompleted: PropTypes.string,
  dateCancelled: PropTypes.string,
  wrapperClassName: PropTypes.string,
};

export default ExperimentDates;
