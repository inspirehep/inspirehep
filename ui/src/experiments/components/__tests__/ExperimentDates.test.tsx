import React from 'react';
import { shallow } from 'enzyme';

import ExperimentDates from '../ExperimentDates';

describe('ExperimentDates', () => {
  it('renders with all props set', () => {
    const wrapper = shallow(
      <ExperimentDates
        dateApproved="1984-02-02"
        dateProposed="1984-02-01"
        dateStarted="1984-02-03"
        dateCancelled="1984-02-04"
        dateCompleted="1984-02-05"
        wrapperClassName="di"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('renders without dateCancelled or dateCompleted', () => {
    const wrapper = shallow(
      <ExperimentDates
        dateApproved="1984-02-02"
        dateProposed="1984-02-01"
        dateStarted="1984-02-03"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('renders with partial dates', () => {
    const wrapper = shallow(
      <ExperimentDates
        dateApproved="1984-02"
        dateProposed="1984"
        dateStarted="1984-02-03"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
