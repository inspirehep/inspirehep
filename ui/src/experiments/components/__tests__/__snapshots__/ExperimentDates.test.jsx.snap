// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperimentDates renders with all props set 1`] = `
<InlineUL
  separator=", "
  wrapperClassName="di"
>
  <span>
    Proposed: 
    Feb 1, 1984
  </span>
  <span>
    Approved: 
    Feb 2, 1984
  </span>
  <span>
    Started: 
    Feb 3, 1984
  </span>
  <span>
    Cancelled: 
    Feb 4, 1984
  </span>
  <span>
    Completed: 
    Feb 5, 1984
  </span>
</InlineUL>
`;

exports[`ExperimentDates renders with partial dates 1`] = `
<InlineUL
  separator=", "
  wrapperClassName={null}
>
  <span>
    Proposed: 
    1984
  </span>
  <span>
    Approved: 
    Feb, 1984
  </span>
  <span>
    Started: 
    Feb 3, 1984
  </span>
  <span>
    Still Running
  </span>
</InlineUL>
`;

exports[`ExperimentDates renders without dateCancelled or dateCompleted 1`] = `
<InlineUL
  separator=", "
  wrapperClassName={null}
>
  <span>
    Proposed: 
    Feb 1, 1984
  </span>
  <span>
    Approved: 
    Feb 2, 1984
  </span>
  <span>
    Started: 
    Feb 3, 1984
  </span>
  <span>
    Still Running
  </span>
</InlineUL>
`;
